pkg/corerp/backend/deployment/deploymentprocessor.go

```diff
@@ -287,6 +287,18 @@ func (dp *deploymentProcessor) Deploy(ctx context.Context, id resources.ID, rend
 
 		err := dp.deployOutputResource(ctx, id, rendererOutput, computedValues, &handlers.PutOptions{Resource: &outputResource, DependencyProperties: deployedOutputResourceProperties})
 		if err != nil {
+			// add the last deployed resource  which was deployed with error to deployed list
+			// cleanup all deployed output resources in reverse order since app deployment has failed
+			outputResource := rpv1.OutputResource{
+				LocalID: outputResource.LocalID,
+				ID:      outputResource.ID,
+			}
+			deployedOutputResources = append(deployedOutputResources, outputResource)
```

Comment:
I can't understand why we need to add the outputResource to a map? The map will only have one item and then the function will exit based on line 302. Am I missing something?